dart/lib/src/http_client/sentry_http_client.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:http/http.dart';
+
+import '../../sentry.dart';
+
+/// A [http](https://pub.dev/packages/http)-package compatible HTTP client
+/// which records requests as breadcrumbs.
+///
+/// Remarks:
+/// If this client is used as a wrapper, a call to close does not close the
+/// given client.
+///
+/// The `SentryHttpClient` can be used as a standalone client like this:
+/// ```dart
+/// import 'package:sentry/sentry.dart';
+///
+/// var client = SentryHttpClient();
+/// try {
+///  var uriResponse = await client.post('https://example.com/whatsit/create',
+///      body: {'name': 'doodle', 'color': 'blue'});
+///  print(await client.get(uriResponse.bodyFields['uri']));
+/// } finally {
+///  client.close();
+/// }
+/// ```
+///
+/// The `SentryHttpClient` can also be used as a wrapper for your own
+/// HTTP [Client](https://pub.dev/documentation/http/latest/http/Client-class.html):
+/// ```dart
+/// import 'package:sentry/sentry.dart';
+/// import 'package:http/http.dart' as http;
+///
+/// final myClient = http.Client();
+///
+/// var client = SentryHttpClient(client: myClient);
+/// try {
+///  var uriResponse = await client.post('https://example.com/whatsit/create',
+///      body: {'name': 'doodle', 'color': 'blue'});
+///  print(await client.get(uriResponse.bodyFields['uri']));
+/// } finally {
+///  client.close();
+/// }
```

Comment:
that's a good insight, what's about adding it to the README?

and even better, what's about adding a button to the sample that does an HTTP request and records a breadcrumb automatically? could be as simple as a GET call to Flutter's website.